Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/CoreDocumentSynchronizationConfig.swift

```diff
@@ -355,16 +272,9 @@ internal class CoreDocumentSynchronization: Hashable {
         }
     }
 
-    func hash(into hasher: inout Hasher) {
-        self.config.hash(into: &hasher)
-    }
-
-    // TODO(STITCH-2329): this linter exception needs to eventually go away
-    // swiftlint:disable force_try
     internal static func filter(forNamespace namespace: MongoNamespace) -> Document {
-        return [CoreDocumentSynchronization.Config.CodingKeys.namespace.rawValue: try! BSONEncoder().encode(namespace)]
+        return [CodingKeys.namespace.rawValue: try? BSONEncoder().encode(namespace) ?? namespace.description]
```

Comment:
[nit] the `?? namespace.description` is unnecessary, and would be incorrect anyway. In the case that encode fails (which it never should), I think it would be better if it was just BSONNull